Wikipedia/Code/WMFLoginViewController.swift

```diff
@@ -13,6 +13,8 @@ class WMFLoginViewController: WMFScrollViewController, UITextFieldDelegate, WMFC
     @IBOutlet fileprivate var loginButton: WMFAuthButton!
     @IBOutlet weak var scrollContainer: UIView!
     
+    public var enableSyncOnLogin: Bool = false
```

Comment:
So rather than giving the login controller knowledge of sync, another possibility is to create a var for a completion block to run.

This way any object that pushes the login VC can run arbitrary code after login completes (or fails). This both generalizes the problem (allows login to trigger many different actions) and encapsulates logic (keeps the login control only concerned with login). 

Thoughts?

Reply:
that sounds good 👍